### PR TITLE
Automatic deployment and hosting on GitHub Pages

### LINKED ISSUE #25: Host the project on GitHub Pages
Set up everything so the state of the main branch can always be seen tested in GitHub Pages

### PR DESCRIPTION
**Pull request by dknigs**
Closes #25 

There is an open question I need your input on, I would be fine either way:
Is it ok to just use https://t-erxleben.github.io/marbellous/demo/frontend/index.html or do we want to add an index.html to the root of our structure to make it accessible from https://t-erxleben.github.io/marbellous/ (with e.g. a redirect)

Also, before we merge I will add a commit removing this branch from the list of branches that trigger the Action, but it's necessary for testing and demonstration.

### CODE REVIEW

**Comment by JBenda** [comment]
Looks great :+1: 
the index.html at top level sounds like the best options
Maybe we can use a [redirect index.html](https://www.w3docs.com/snippets/html/how-to-redirect-a-web-page-in-html.html)
Then we prevent awkward GUI placement with the iframe

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        run: emcc -v
+
+      - name: Tree
+        run: tree
+
+      - name: Run emCMake
+        run: |
+          cd demo/backend/
+          emcmake cmake
+          make install
```
Comment (t-erxleben) [inline .github/workflows/main.yml:32]: According to the [docs](https://emscripten.org/docs/compiling/Building-Projects.html) `make` should also be wrapped. Obviously it builds as it is but this should be more fail safe.

**Review by t-erxleben — APPROVED** [review]
Nice. Redirecting seems like the best option.

**Review by JBenda — APPROVED** [review]
Nice

**Pull request merged**